database/rrdset.c

```diff
@@ -1089,6 +1091,65 @@ inline void rrdset_next_usec(RRDSET *st, usec_t microseconds) {
 // ----------------------------------------------------------------------------
 // RRDSET - process the collected values for all dimensions of a chart
 
+void rrdset_dump_debug_state(RRDSET *st) {
+#ifdef NETDATA_INTERNAL_CHECKS
+    if (debug_flags & D_REPLICATION) {
+        RRDDIM *rd;
+        debug(D_REPLICATION, "Chart state %s: counter=%zu counter_done=%zu current_entry=%ld usec_since_last=%llu last_updated=%ld.%ld"
+                             " last_collected=%ld.%ld collected_total=%lld last_collected_total=%lld",
+                             st->id, st->counter, st->counter_done, st->current_entry, st->usec_since_last_update,
+                             st->last_updated.tv_sec, st->last_updated.tv_usec,
+                             st->last_collected_time.tv_sec, st->last_collected_time.tv_usec,
+                             st->collected_total, st->last_collected_total);
+        netdata_rwlock_rdlock(&st->rrdset_rwlock);
+        rrddim_foreach_read(rd, st) {
+            debug(D_REPLICATION, "Dimension state %s.%s: calculated_value=" CALCULATED_NUMBER_FORMAT
+                                 " last_calculated_value=" CALCULATED_NUMBER_FORMAT
+                                 " last_stored_value=" CALCULATED_NUMBER_FORMAT
+                                 " collected_value=" COLLECTED_NUMBER_FORMAT
+                                 " last_collected_value=" COLLECTED_NUMBER_FORMAT
+                                 " col_counter=%zu"
+                                 " col_volume=" CALCULATED_NUMBER_FORMAT
+                                 " store_volume=" CALCULATED_NUMBER_FORMAT
+                                 " last_coll_time=%ld.%ld",
+                                 st->id, rd->id, rd->calculated_value, rd->last_calculated_value,
+                                 rd->last_stored_value, rd->collected_value, rd->last_collected_value,
+                                 rd->collections_counter, rd->collected_volume,
+                                 rd->stored_volume, rd->last_collected_time.tv_sec, rd->last_collected_time.tv_usec);
+            #ifdef ENABLE_DBENGINE
+            if (st->rrd_memory_mode == RRD_MEMORY_MODE_DBENGINE && rd->state->handle.rrdeng.descr) {
```

Comment:
```suggestion
            if (st->rrd_memory_mode == RRD_MEMORY_MODE_DBENGINE && !rrddim_flag_check(rd, RRDDIM_FLAG_ARCHIVED) && rd->state->handle.rrdeng.descr) {
```
I can't reproduce the crash for some reason, but this is one potential problem I can think of.

Reply:
This was it, confirmed that it is is fixed.